cmd/jujud/agent/machine/manifolds.go

```diff
@@ -784,6 +786,14 @@ func commonManifolds(config ManifoldsConfig) dependency.Manifolds {
 				Logger:        loggo.GetLogger("juju.worker.secretbackendsrotate"),
 			},
 		))),
+
+		// The controlsocket worker runs on the controller machine.
+		controlSocketName: ifPrimaryController(controlsocket.Manifold(controlsocket.ManifoldConfig{
```

Comment:
what does "primary" refer to? I was thinking it meant the leader in HA

Reply:
So that does mean it's the leader, my worry is there might be a race when switching leaders that the endpoints won't be up in time, so you'll get a flurry of errors. This might end up being a sequencing problem. I think it's much simpler to just have the endpoints there all the time. You won't get the issue of endpoints not being there, and fighting with putting up and down the server.

Reply:
Ok fair point